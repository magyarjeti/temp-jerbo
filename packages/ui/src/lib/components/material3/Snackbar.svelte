<script lang="ts">
	import JetiIcon from '../JetiIcon.svelte';
	import { fade } from 'svelte/transition';

	interface Props {
		message?: string | null;
	}

	let { message = $bindable() }: Props = $props();

	function clear() {
		message = undefined;
	}

	$effect(() => {
		setTimeout(() => {
			clear();
		}, 10000);
	});
</script>

{#if message}
	<div class="fixed bottom-[4dvh] w-full">
		<div class="container flex justify-center">
			<div
				transition:fade
				class="typo-body-medium flex min-h-12 w-fit items-center bg-inverse-surface py-2 pl-4 pr-3 text-inverse-on-surface elevation-3 shape-small"
			>
				<p>{message}</p>
				<button onclick={clear} aria-label="bezárás" class="ml-3 h-6 w-6 cursor-pointer"
					><JetiIcon>close</JetiIcon></button
				>
			</div>
		</div>
	</div>
{/if}
