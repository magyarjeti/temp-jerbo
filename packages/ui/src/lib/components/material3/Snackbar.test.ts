import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import Snackbar from './Snackbar.svelte';

// TODO: test behavior
test('Snackbar snapshot', () => {
	const { container } = render(Snackbar);
	expect(container).toMatchSnapshot();
});
