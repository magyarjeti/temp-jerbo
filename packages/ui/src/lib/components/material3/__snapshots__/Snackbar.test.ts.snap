// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snackbar snapshot 1`] = `
<div>
  
</div>
`;
